File: leetcode/0520_detect_capital.kt
package leetcode

class Solution {
    fun detectCapitalUse(word: String): Boolean {
        if (word.all({ char -> char.isUpperCase() })) {
            return true
        } else if (word.all({ char -> char.isLowerCase() })) {
            return true
        } else if (word[0].isUpperCase() && word.substring(1).all({ char -> char.isLowerCase() })) {
            return true
        } else {
            return false
        }
    }
}

fun main(args: Array<String>) {
    val solution = Solution()

    assert(solution.detectCapitalUse("USA"))
    assert(solution.detectCapitalUse("leetcode"))
    assert(solution.detectCapitalUse("Google"))
    assert(!solution.detectCapitalUse("FlaG"))
}

File: leetcode/0520_detect_capital.py
# -*- coding: utf-8 -*-


class Solution:
    def detectCapitalUse(self, word):
        if all(char.isupper() for char in word):
            return True
        elif all(char.islower() for char in word):
            return True
        elif word[0].isupper() and all(char.islower() for char in word[1:]):
            return True
        return False


if __name__ == '__main__':
    solution = Solution()

    assert solution.detectCapitalUse('USA')
    assert solution.detectCapitalUse('leetcode')
    assert solution.detectCapitalUse('Google')
    assert not solution.detectCapitalUse('FlaG')
